src/EditorFeatures/Core/CommandHandlers/AbstractGoToCommandHandler.cs

```diff
@@ -47,7 +47,7 @@ public CommandState GetCommandState(TCommandArgs args)
             var findUsagesService = document?.GetLanguageService<TLanguageService>();
             return findUsagesService != null
                 ? CommandState.Available
-                : CommandState.Unavailable;
+                : CommandState.Unspecified;
```

Comment:
Yes, the command becomes unavailable to the LSP client.

As to avoid it... I'm not sure?  Seems to be case by case unless we always want to use unspecified (in that case... analyzer?)